src/libstd/io/mod.rs

```diff
@@ -219,6 +318,29 @@ pub trait Read {
     ///
     /// The returned adaptor also implements `Read` and will simply borrow this
     /// current reader.
+    ///
+    /// # Examples
+    ///
+    /// [`File`][file]s implement `Read`:
+    ///
+    /// [file]: ../std/fs/struct.File.html
+    ///
+    /// ```
+    /// use std::io;
+    /// use std::fs::File;
+    /// use std::io::Read;
+    ///
+    /// # fn foo() -> io::Result<()> {
+    /// let mut f = try!(File::open("foo.txt"));
+    /// let mut buffer = String::new();
+    ///
+    /// let reference = f.by_ref();
```

Comment:
so this example looks good, but it introduces some weirdness, with type annotations. check it out, i pushed it up.